src/map/map.ts

```diff
@@ -1200,53 +1258,133 @@ export class PropertiesMap {
+                  'y': this._yValues(1),
+                  'z': this._zValues(1),
+
+                  'marker.symbol': symbol,
+              } as Data, 1);
+          } else {
+              const xaxis = this._plot._fullLayout.xaxis;
+              const yaxis = this._plot._fullLayout.yaxis;
+
+              const computeX = (data: number) => xaxis.l2p(data) + xaxis._offset;
+              const computeY = (data: number) => yaxis.l2p(data) + yaxis._offset;
+
+              const x = computeX(this._xValues(0)[0][selected]);
+              const y = computeY(this._yValues(0)[0][selected]);
+
+              // hide the point if it is outside the plot, allow for up to 10px
+              // overflow (for points just on the border)
+              const minX = computeX(xaxis.range[0]) - 10;
+              const maxX = computeX(xaxis.range[1]) + 10;
+              const minY = computeY(yaxis.range[1]) - 10;
+              const maxY = computeY(yaxis.range[0]) + 10;
+              if (!isFinite(x) || !isFinite(y) || x < minX || x > maxX || y < minY || y > maxY) {
+                  marker.style.display = 'none';
+              } else {
+                  marker.style.display = 'block';
+              }
+
+              // -10 since we want the centers to match, and the marker div is 20px wide
+              marker.style.top = `${y - 10}px`;
+              marker.style.left = `${x - 10}px`;
             }
 
-            this._restyle({
-                'x': this._xValues(1),
-                'y': this._yValues(1),
-                'z': this._zValues(1),
+          const indexes = this._indexer.from_environment(this._currentList[guid]);
+          this.onselect(indexes, guid);
+        }
+      }
+
+    private _addMarker(guid: string, index: number = 0, color?: string) {
+      let marker = document.getElementById(`chsp-selected-${guid}`);
+      if (!this._is3D()) {
+        if (marker === null) {
+          const activeButton = document.getElementById(`chsp-activate-${guid}`);
+          if (activeButton !== null) {
+            marker = document.createElement('div');
+            marker.classList.add('chsp-selected', 'chsp-inactive-map');
+            if (guid === this._active) {
+              marker.classList.toggle('chsp-inactive-map', false);
+              marker.classList.toggle('chsp-active', true);
+            }
+            marker.id = `chsp-selected-${guid}`;
+            marker.onclick = () => {this._setActive(guid); };
+
+            color = activeButton.style.backgroundColor;
+            marker.style.backgroundColor = color;
+            this._root.appendChild(marker);
+          }
+        } else {this._root.appendChild(marker); }
+      }
+      if (! this._guids.includes(guid)) {
+        this._guids.push(guid);
+      }
+      this._currentList[guid] = Math.max(0, index);
+    }
+    private _removeMarker(guid: string, removeGUID: boolean = true) {
+        const marker = document.getElementById(`chsp-selected-${guid}`);
```

Comment:
Is there a need (i.e. is getting a time-consuming operation)?

Reply:
It is not so much time consuming (although it might be, I'll have to look into this), but rather making sure there is a single point of knowledge for each element in the system. In turn, I find this makes the code easier to understand and modify later (we have less places to remember to modify).